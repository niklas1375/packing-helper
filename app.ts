import express from "express";
import session from "express-session";
import "dotenv/config";
import { sessionSecret } from "./modules/secret-config";

import * as routes from "./routes";

const app = express();

app.use(express.json());

app.use(
  session({
    secret: sessionSecret,
    resave: false,
    saveUninitialized: false,
    cookie: {
      secure: "auto",
      maxAge: 3600000,
    },
  })
);

const router = express.Router();
routes.register(router);
app.use("/api", router);
app.get("/", (_, res) => {
  res.json({
    message: "API is up and running.",
  });
});

export default app;
