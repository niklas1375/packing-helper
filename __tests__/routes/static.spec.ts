import request from "supertest";

import app from "../../app";

import static_accomodation from "../json/static/accomodation.json";
import static_activities from "../json/static/activities.json";
import static_transport from "../json/static/transport.json";
import static_triptypes from "../json/static/triptypes.json";
import static_weather from "../json/static/weather.json";

describe("Static routes", () => {
  test("Get all accomodations", async () => {
    const res = await request(app).get("/api/accomodation");
    expect(res.body).toEqual(static_accomodation);
  });
  test("Get all activities", async () => {
    const res = await request(app).get("/api/activities");
    expect(res.body).toEqual(static_activities);
  });
  test("Get all transports", async () => {
    const res = await request(app).get("/api/transport");
    expect(res.body).toEqual(static_transport);
  });
  test("Get all triptypes", async () => {
    const res = await request(app).get("/api/triptypes");
    expect(res.body).toEqual(static_triptypes);
  });
  test("Get all weathers", async () => {
    const res = await request(app).get("/api/weather");
    expect(res.body).toEqual(static_weather);
  });
});
