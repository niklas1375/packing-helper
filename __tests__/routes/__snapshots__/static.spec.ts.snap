// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Static routes Get all accomodations 1`] = `
[
  {
    "key": "cm0167zsu001wdlnegyn5cv98",
    "name": "Camping",
  },
  {
    "key": "cm0167zsz004xdlnespbj4rcp",
    "name": "Hotel",
  },
  {
    "key": "cm0167zsu0003dlne9o9ahwl1",
    "name": "HÃ¼tte",
  },
  {
    "key": "cm0167zsx004bdlnefrf1clkm",
    "name": "Jugendherberge",
  },
]
`;

exports[`Static routes Get all activities 1`] = `
[
  {
    "key": "cm0167ztk007cdlnesmlwzr3r",
    "name": "Bouldern",
  },
  {
    "key": "cm0167ztj006xdlneo58kcn7h",
    "name": "Fahrrad fahren",
  },
  {
    "key": "cm0167zsu0001dlnehd63mfi8",
    "name": "Klettersteig",
  },
  {
    "key": "cm0167ztl007kdlnexvzg32ep",
    "name": "Konzert",
  },
  {
    "key": "cm0167zsu002ddlne3q0dftko",
    "name": "Laufen",
  },
  {
    "key": "cm0167zsu000ddlneh43thcsy",
    "name": "Rudern",
  },
  {
    "key": "cm0167zsu000tdlneyfddgasz",
    "name": "Sightseeing",
  },
  {
    "key": "cm0167zsy004gdlne9ladyn7e",
    "name": "Ski fahren",
  },
  {
    "key": "cm0167zsu0000dlnewnnz6afq",
    "name": "Strand",
  },
  {
    "key": "cm0167zsu0002dlned2usb8ha",
    "name": "Tauchen",
  },
  {
    "key": "cm0167zsu001idlneoqabzsi8",
    "name": "Wandern",
  },
]
`;

exports[`Static routes Get all transports 1`] = `
[
  {
    "key": "cm0167zt10057dlnebb9xxnn4",
    "name": "Auto",
  },
  {
    "key": "cm0167zt00050dlnewtxq8mqe",
    "name": "Bus",
  },
  {
    "key": "cm0167zt2005adlnecyn33jm8",
    "name": "Flugzeug",
  },
  {
    "key": "cm0167zt5005hdlnephzk0wzi",
    "name": "Zug",
  },
]
`;

exports[`Static routes Get all triptypes 1`] = `
[
  {
    "key": "cm0167zt8005odlnepqzqsod3",
    "name": "Business",
  },
  {
    "key": "cm0167zt9005xdlne4bola1zi",
    "name": "Freizeit",
  },
  {
    "key": "cm0167zta0061dlnex56t1ejl",
    "name": "Mix",
  },
]
`;

exports[`Static routes Get all weathers 1`] = `
[
  {
    "key": "cold",
    "name": "Kaltes Wetter",
  },
  {
    "key": "wet",
    "name": "Nasses Wetter",
  },
  {
    "key": "sunny",
    "name": "Sonniges Wetter",
  },
  {
    "key": "warm",
    "name": "Warmes Wetter",
  },
]
`;
